component-tools/src/main/java/org/talend/sdk/component/tools/DocBaseGenerator.java

```diff
@@ -293,6 +342,119 @@ private String doResolveProperty(final String propPath, final String paramRef) {
+                    // uiInfo.addNestedLayouts(subLayouts);
+                }
+            }
+
+            return layouts;
+        }
+
+        private String getLayout(final ParameterMeta parent, final String layout, final ParameterMeta param) {
+            if (parent == null) {
+                return "tcomp::ui::gridlayout::Main::value";
+            }
+
+            List<String> main = getPropertiesByLayout(parent, "tcomp::ui::gridlayout::Main::value");
+            List<String> advanced = getPropertiesByLayout(parent, "tcomp::ui::gridlayout::Advanced::value");
+
+            return advanced.contains(param.getName()) ? "tcomp::ui::gridlayout::Advanced::value"
+                    : main.contains(param.getName()) ? "tcomp::ui::gridlayout::Main::value" : layout;
+        }
+
+        private List<String> getPropertiesByLayout(final ParameterMeta parent, final String layout) {
+            String layoutConfig = parent.getMetadata().get(layout);
+            List<String> properties;
+            if (layoutConfig != null) {
+                properties = Collections
+                        .list(new StringTokenizer(layoutConfig, "|"))
+                        .stream()
+                        .map(token -> (String) token)
+                        .collect(Collectors.toList());
+            } else {
+                properties = Collections.emptyList();
+            }
+
+            return properties;
+        }
+
+        Map<String, Map<String, List<Param>>> getParametersWithUInfo() {
+            final Map<String, Map<String, List<Param>>> params = new HashMap<>();
+            final Map<String, UIInfo> uiInfos = getUIParamByPath(parameters);
+
+            for (Param p : toParams()) {
+                setParametersWithUInfoRecurs(p, uiInfos, params);
+            }
+
+            return params;
+        }
+
+        void setParametersWithUInfoRecurs(final Param p, final Map<String, UIInfo> uiInfos,
+                final Map<String, Map<String, List<Param>>> params) {
+            UIInfo info = uiInfos.get(p.fullPath);
+            p.setUiInfo(info);
+
+            Map<String, List<Param>> section = params.get(info.getSection());
+            if (section == null) {
+                section = new HashMap<>();
+                params.put(info.getSection(), section);
+            }
+
+            for (String l : info.getNestedLayout()) {
+                List<Param> layout = section.get(l);
+                if (layout == null) {
+                    layout = new ArrayList<>();
+                    section.put(l, layout);
+                }
+
+                layout.add(p);
+            }
+
+            for (Param n : p.getNested()) {
+                setParametersWithUInfoRecurs(n, uiInfos, params);
+            }
+        }
+
+        private Collection<Param> toParams() {
+            Collection<Param> params = new ArrayList<>();
+            for (ParameterMeta p : parameters) {
```

Comment:
Please dont duplicate the findDefault method, ensure there is a single method responsible for it to let us enrich the logic without duplicating it (typically ui defaults support)

Reply:
Refarctored.